test/integration/dockerregistryclient_test.go

```diff
@@ -4,11 +4,61 @@ package integration
 
 import (
 	"reflect"
+	"regexp"
 	"testing"
+	"time"
 
 	"github.com/openshift/origin/pkg/dockerregistry"
 )
 
+const (
+	pulpRegistryName        = "registry.access.redhat.com"
+	dockerHubV2RegistryName = "index.docker.io"
+	dockerHubV1RegistryName = "registry.hub.docker.com"
+	quayRegistryName        = "quay.io"
+
+	maxRetryCount = 4
+	retryAfter    = time.Millisecond * 500
+)
+
+var reUnreachableErr *regexp.Regexp
+
+func init() {
+	reUnreachableErr = regexp.MustCompile(`(?:dial|read) tcp|net/http|could not be reached`)
```

Comment:
heh, might need a test for the test.  I can definitely see the intent, but I can't say with certainty that this fulfills the intent.  @liggitt can you read it?